packages/flutter_driver/lib/src/extension/extension.dart

```diff
@@ -98,19 +150,29 @@ typedef CommandHandlerCallback = Future<Result> Function(Command c);
 /// Signature for functions that deserialize a JSON map to a command object.
 typedef CommandDeserializerCallback = Command Function(Map<String, String> params);
 
-/// Signature for functions that run the given finder and return the [Element]
-/// found, if any, or null otherwise.
-typedef FinderConstructor = Finder Function(SerializableFinder finder);
+/// Used to expand the new Finder
+abstract class FinderExtension {
```

Comment:
This whole class appears to be unnecessary. After reading the example above, I thikn it can be simplified down to just have the user extend `SerializableFinder` and provide the right `DeserializeFinderFactory` that creates the expected `SerializableFinder`. See my comment below on `_createFinder`.

If we can do that, this whole class goes away and eliminates some complexity.

Reply:
But createFinder cannot be put together with SerializableFinder, otherwise it will easily lead to indirect reference to "dart:ui"", and reference to "dart:ui" in the test code will cause it to fail to run

Reply:
Ahh ok, that makes sense.